src/main/java/com/aws/greengrass/mqtt/bridge/clients/MQTTClient.java

```diff
@@ -162,9 +162,18 @@ public void start() throws MessageClientException {
         connectAndSubscribe();
     }
 
-    private void disconnect() {
-        // 0ms quiescence time, just send the disconnect packet immediately
-        disconnect(0);
+    @SuppressWarnings({"PMD.AvoidCatchingGenericException", "PMD.AvoidCatchingNPE"})
+    private void disconnectForcibly() {
+        try {
+            long doNotQuiesce = 0L;
+            long doNotWaitForDisconnectPacket = 1L; // since 0L means no timeout
+            mqttClientInternal.disconnectForcibly(doNotQuiesce, doNotWaitForDisconnectPacket);
+        } catch (MqttException e) {
+            LOGGER.atWarn().cause(e).log("Unable to disconnect forcibly");
+        } catch (NullPointerException ignore) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception has been ignored. The catch-block hides information about the stack trace. Not logging or forwarding the stack trace of an exception can mask unexpected errors and complicated debugging.

**Fix**: Use a catch block to handle the exception and write a log message declaring that the exception happened. 
[Learn more.](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)

Reply:
https://github.com/aws-greengrass/aws-greengrass-mqtt-bridge/pull/162#discussion_r1380909172